packages/flutter/lib/src/gestures/velocity_tracker.dart

```diff
@@ -231,19 +231,16 @@ class VelocityTracker {
     } while (sampleCount < _historySize);
 
     if (sampleCount >= _minSampleSize) {
-      final LeastSquaresSolver xSolver = LeastSquaresSolver(time, x, w);
-      final PolynomialFit? xFit = xSolver.solve(2);
-      if (xFit != null) {
-        final LeastSquaresSolver ySolver = LeastSquaresSolver(time, y, w);
-        final PolynomialFit? yFit = ySolver.solve(2);
-        if (yFit != null) {
-          return VelocityEstimate( // convert from pixels/ms to pixels/s
-            pixelsPerSecond: Offset(xFit.coefficients[1] * 1000, yFit.coefficients[1] * 1000),
-            confidence: xFit.confidence * yFit.confidence,
-            duration: newestSample.time - oldestSample.time,
-            offset: newestSample.point - oldestSample.point,
-          );
-        }
+      late final PolynomialFit? xFit = LeastSquaresSolver(time, x, w).solve(2);
+      late final PolynomialFit? yFit = LeastSquaresSolver(time, y, w).solve(2);
+
+      if (xFit != null && yFit != null) {
```

Comment:
This is possibly less performant because it now solves yFit even if xFit is null. Probably not a big deal...

@gspencergoog (looking at git blame) do you have any thoughts on whether this slight performance change will matter at all?

Reply:
> This is possibly less performant because it now solves `yFit` even if `xFit` is null.

Actually, that's what the [`late`](https://dart.dev/language/variables#late-variables) is for! (If something is marked as `late`, it won't be evaluated until it's referenced somewhere else.)

Reply:
That's a really interesting usage of `late`!

Reply:
Can we add a short comment to explain this?

Reply:
> Can we add a short comment to explain this?

That's a great idea!